src/components/DateTimePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) DD360 and its affiliates.
+ */
+import {
+  CSSProperties,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState
+} from 'react'
+import { CalendarIcon } from '@heroicons/react/outline'
+import { composeClasses } from 'lib/classes'
+import BaseInput, { InputProps } from 'components/Form/Input/BaseInput'
+import Calendar from './Calendar'
+import Clock from './Clock'
+
+export interface IDateTimePicker {
+  minDate?: Date
+  maxDate?: Date
+  readonly style?: CSSProperties
+  readonly className?: string
+  readonly handleDateChange?: (date: Date) => void
+  readonly pickerClassName?: string
+}
+
+export type DateTimePickerProps = IDateTimePicker & InputProps
+
+function formatInput(date: string) {
+  let inputText = date
+  if (inputText.length > 2 && inputText.charAt(2) !== '/') {
+    inputText =
+      inputText.slice(0, 2) + '/' + inputText.slice(2, inputText.length)
+  }
+  if (inputText.length > 5 && inputText.charAt(5) !== '/') {
+    inputText =
+      inputText.slice(0, 5) + '/' + inputText.slice(5, inputText.length)
+  }
+  if (inputText.length > 10 && inputText.charAt(10) !== ' ') {
+    inputText =
+      inputText.slice(0, 10) + ' ' + inputText.slice(10, inputText.length)
+  }
+  if (inputText.length > 13 && inputText.charAt(13) !== ':') {
+    inputText =
+      inputText.slice(0, 13) + ':' + inputText.slice(13, inputText.length)
+  }
+  if (inputText.length > 16 && inputText.charAt(16) !== ' ') {
+    inputText =
+      inputText.slice(0, 16) + ' ' + inputText.slice(16, inputText.length)
+  }
+
+  return inputText.slice(0, 19).toUpperCase()
+}
+
+function validateDate(date: Date, minDate?: Date, maxDate?: Date) {
+  if (minDate && maxDate)
+    return (
+      date.valueOf() >= minDate?.valueOf() &&
+      date.valueOf() <= maxDate?.valueOf()
+    )
+  if (minDate) return date.valueOf() >= minDate?.valueOf()
+  if (maxDate) return date.valueOf() <= maxDate?.valueOf()
+  return true
+}
+
+const DateTimePicker = ({
+  className,
+  value,
+  onChange,
+  variant,
+  isDisabled,
+  handleDateChange,
+  pickerClassName,
+  minDate,
+  maxDate,
+  ...props
+}: DateTimePickerProps) => {
+  const containerRef = useRef<HTMLDivElement>(null)
+  const [showDateTimePicker, setShowDateTimePicker] = useState<boolean>(false)
+  const [dateTime, setDateTime] = useState(formatInput(String(value ?? '')))
+  const currentDate = useMemo(() => {
+    const [date, time, amPm] = dateTime.split(' ')
+    const dateArray = date.split('/')
+
+    let newDate = undefined
+    let timeFormat = '00:00'
+
+    if (dateArray.length === 3) {
```

Comment:
I believe that it is a good idea to add a validation if the value doesn't contain the expected type, just to prevent any unexpected case